pkg/client/events/replay_client.go

```diff
@@ -27,19 +27,47 @@ const (
 	// eventsBytesRetryDelay is the delay between retry attempts when the events
 	// bytes observable returns an error.
 	eventsBytesRetryDelay = time.Second
+)
+
+// ConnStateStatus represents the status of the underlying websocket connection
+type ConnStateStatus int
```

Comment:
I'm beginning to get a bit confused why all of this is in `pkg/client/events/replay_client.go`?

Reply:
This is just for a stateful connection (websocket)

Reply:
THought on naming it accordignly 

e.g. `StatefulConnectionStatus` or `WebsocketConnectionState`

Reply:
Moved to a dedicated `websocket/state.go` file